tensorflow_lite_support/python/task/audio/core/tensor_audio.py

```diff
@@ -95,26 +99,48 @@ def load_from_audio_record(self, record: audio_record.AudioRecord) -> None:
     data = record.read(self._buffer_size)
     self.load_from_array(data.astype(np.float32))
 
-  def load_from_array(self, src: np.ndarray) -> None:
-    """Loads audio data from a NumPy array.
+  def load_from_array(self,
+                      src: np.ndarray,
+                      offset: int = None,
+                      size: int = None) -> None:
+    """Loads the audio data from a NumPy array.
 
     Args:
-      src: A NumPy array contains the input audio.
-
+      src: A NumPy source array contains the input audio.
+      offset: An optional offset for loading a slice of the `src` array to the
+        buffer.
+      size: An optional size parameter denoting the number of samples to load
+        from the `src` array.
     Raises:
-      ValueError if the input audio is too large or if it contains an invalid
-      number of channels.
+      ValueError: Raised if the input array has an incorrect shape or if
+      offset + size exceeds the length of the `src` array.
     """
-    if len(src) != len(self._buffer):
-      raise ValueError(
-        f"Input audio contains an invalid number of samples. "
-        f"Expect {len(self._buffer)}.")
-    elif src.shape[1] != self._format.channels:
+    if src.shape[1] != self._format.channels:
       raise ValueError(
         f"Input audio contains an invalid number of channels. "
         f"Expect {self._format.channels}.")
 
-    self._buffer = src
+    if size is None:
+      size = len(src)
+    if offset is None:
+      offset = 0
+
+    if offset + size > len(src):
+      raise ValueError(
+        f"Index out of range. offset {offset} + size {size} should be <= "
+        f"src's length: {len(src)}")
+
+    if len(src) > len(self._buffer):
```

Comment:
Shouldn't this be `if len(src) >= len(self._buffer)`?

Reply:
@khanhlvg Uh no because the opposite case is `<= newData.length` [here](https://github.com/tensorflow/tflite-support/blob/e109aa0094e6fd5bb1e30c61ec9b19194c452ac3/tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/audio/TensorAudio.java#L304).